### PR TITLE
Initialize db before serving over HTTP

### PR DESCRIPTION
**Pull request by kshakir**

### CODE REVIEW

File: engine/src/main/scala/cromwell/server/CromwellSystem.scala [hunk 1]
```diff
@@ -5,11 +5,21 @@ import akka.http.scaladsl.Http
...
 import scala.concurrent.Future
 
 trait CromwellSystem {
+  /*
+  Initialize the services-store (aka a collection of DAOs), if they haven't been already.
+  This is the closest we have to:
+  - https://github.com/broadinstitute/firecloud-orchestration/blob/3c9482b/DEVNOTES.md
+  - https://github.com/broadinstitute/clio/blob/7253ec0/clio-server/src/main/scala/org/broadinstitute/clio/server/service/ServerService.scala#L58-L66
+  - https://en.wikipedia.org/wiki/Data_access_object
+   */
```
Comment (mcovarr) [inline engine/src/main/scala/cromwell/server/CromwellSystem.scala]: Maybe a comment as to the problem that is avoided by this being here, lest someone be tempted to refactor it to a spot where it no longer has the desired effect at the desired point in the initialization?

**Comment by mcovarr** [comment]
:+1:

File: engine/src/main/scala/cromwell/server/CromwellSystem.scala (hunk 1 above)
Comment (kshakir) [inline engine/src/main/scala/cromwell/server/CromwellSystem.scala]: Added above

**Comment by Horneth** [comment]
👍

**Pull request merged**